### PR TITLE
ClientTlsContext tests and rename-o-rama

### PR DESCRIPTION
**Pull request by graebm**
The arguments to these functions confused me so I renamed them.
Will Captain Consistency come and kill me in my dreams now?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/resources/unittests.crt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+-----BEGIN CERTIFICATE-----
```
Comment (justinboswell) [inline test/resources/unittests.crt]: Are these just random certs to see if the parser/IO works? They don't have any permissions anywhere right?

**Comment by graebm** [comment]
> Does elasticurl.py need to be updated well?

nope. It was passing the arguments positionally

File: test/resources/unittests.crt (hunk 1 above)
Comment (graebm) [reply test/resources/unittests.crt]: I copied them from aws-c-io's unit tests.
The unit tests I have so far are just testing the parser,
but when I get around to writing tests that make actual connections, we'll use them in similar ways to how they're used in aws-c-io

Comment (graebm) [reply test/resources/unittests.crt]: I assume they have no permissions, since they're publicly in aws-c-io

**Pull request merged**